### PR TITLE
Extracted the "Validation Groups" book contents to a dedicated cookbook tutorial

### PR DESCRIPTION
**Pull request by javiereguiluz**
This is the first step to implement the ideas explained in #6093 to streamline the book chapters about Forms and Validation.

---

99% of this PR is just moving content locations. The "new" contents are just two phrases to introduce and "glue" things. This will be easy to review. Thanks!

### CODE REVIEW

File: book/forms.rst [hunk 1]
```diff
@@ -463,173 +463,15 @@ Validation is a very powerful feature of Symfony and has its own
...
-
-Note that when you do that, the form will still run basic integrity checks,
-for example whether an uploaded file was too large or whether non-existing
-fields were submitted. If you want to suppress validation, you can use the
-:ref:`POST_SUBMIT event <cookbook-dynamic-form-modification-suppressing-form-validation>`.
-
-.. index::
-   single: Forms; Validation groups based on submitted data
-
 .. _book-form-validation-groups:
```
Comment (xabbuh) [inline book/forms.rst]: We should add labels for all other removed headlines to make the reader aware of where to find the moved information.

Comment (javiereguiluz) [inline book/forms.rst]: In 6017fc9 I've added all the missing labels. Please review them. Thanks!

File: book/forms.rst [hunk 2]
```diff
@@ -463,173 +463,18 @@ Validation is a very powerful feature of Symfony and has its own
...
-        ->add('previousStep', 'submit', array(
-            'validation_groups' => false,
-        ))
-        ->getForm();
+.. tip::
 
-Now the form will skip your validation constraints. It will still validate
-basic integrity constraints, such as checking whether an uploaded file was too
-large or whether you tried to submit text in a number field.
+    Read :doc:`/cookbook/validation/validation_groups` article to learn how to
```
Comment (xabbuh) [inline book/forms.rst]: Read the [...]

File: book/forms.rst [hunk 3]
```diff
@@ -463,173 +463,18 @@ Validation is a very powerful feature of Symfony and has its own
...
+    select dynamically the constraints to apply for a given form (e.g. to
```
Comment (xabbuh) [inline book/forms.rst]: [...] how to dynamically select [...]

**Comment by xabbuh** [comment]
👍 (just two last minor comments)

**Comment by javiereguiluz** [comment]
Fixed the last issues. Thanks!

**Comment by xabbuh** [comment]
@javiereguiluz Looks like you need to push your changes. :)

**Comment by javiereguiluz** [comment]
@xabbuh 😊  it's done now. Thanks.

**Comment by xabbuh** [comment]
👍

**Comment by WouterJ** [comment]
This article now jumps a bit between validation, form and validation. I don't really like that. I think everything about forms (including "Disabling Validation", "Validation Groups based on the Submitted Data" and "Validation Groups based on the Clicked Button") should be put under one section (e.g. "Validation Groups and Forms"). This would make it easier to search for and link to.

**Comment by javiereguiluz** [comment]
@WouterJ closing this PR then.

**Comment by WouterJ** [comment]
@javiereguiluz I think you understood me wrong. I like these changes, what I don't like is that there now is no clear boundary between validation and form: All sections are top level sections.

What I propose is to restructure the article like this:
```
Validation Groups
  +- <little introduction, like now>
  +- Using Validation Groups in Forms
  |   +- <introduction again, the other half>
  |   +- Disabling Validation
  |   +- Validation Groups based on the Submitted Data
  |   +- Validation Groups based on the Clicked Button
  +- Validation Group Sequence
  +- Group Sequence Providers
```

File: cookbook/validation/validation_groups.rst [hunk 4]
```diff
@@ -0,0 +1,655 @@
...
+        ));
+    }
+
+Note that when you do that, the form will still run basic integrity checks,
+for example whether an uploaded file was too large or whether non-existing
+fields were submitted. If you want to suppress validation, you can use the
+:ref:`POST_SUBMIT event <cookbook-dynamic-form-modification-suppressing-form-validation>`.
+
+Validation Groups based on the Submitted Data
+----------------------------------------------
```
Comment (WouterJ) [inline cookbook/validation/validation_groups.rst]: one `-` too much here

File: cookbook/validation/validation_groups.rst [hunk 5]
```diff
@@ -0,0 +1,655 @@
...
+                    return array('Default', 'person');
+                }
+
+                return array('Default', 'company');
+            },
+        ));
+    }
+
+Validation Groups based on the Clicked Button
+----------------------------------------------
```
Comment (WouterJ) [inline cookbook/validation/validation_groups.rst]: same here

**Comment by bogdaniel** [comment]
Maybe you ahould add some info about adding.javascript validation / show errors combined with sf validation.

**Comment by javiereguiluz** [comment]
Closing as won't fix because the validation groups have already been moved to their own article: http://symfony.com/doc/current/validation/groups.html

**Pull request closed without merging**